esx_service/vmdk-opsd

```diff
@@ -112,32 +113,65 @@ stop() {
    if [ -n "${PID}" ]; then
       echo "Stopping $OPSD_TAG with PID=${PID}"
       kill -$SIGTERM ${PID}
+
       local COUNT=0
       while kill -0 ${PID} > /dev/null 2>&1; do
          sleep ${WAIT_KILL_SEC};
+         echo "Waiting for operations in flight to complete, please do not interrupt..."
          if [ $((COUNT++)) -gt ${NUM_RETRY} ]; then
-            echo "Sending kill -${SIGKILL} to ${PID} after kill -${SIGTERM} did not stop ${PID}"
-            kill -${SIGKILL} ${PID}
+            echo "Kill -${SIGTERM} did not stop ${PID}"
             break
          fi
       done
    fi
 
+   cleanup_resource
+   if [ $? -ne 0 ]; then
+      exit $?
+   fi
+
+   check_running_status
+   if [ $? -eq 0 ]; then
+      echo "Failed to stop $OPSD_TAG. Please try shutdown to force kill the process"
```

Comment:
nit: `shutdown` feels like a normal shutdown (i.e. with draining in flight ops.
The verb should be ""shutdown" (or "stop") for normal op. and "kill" or "forcestop" for forced op

Reply:
"kill" does not work - the logic will get stuck in an infinite nesting because the linux system call kill will be invoked. I can use 'forcestop' if you wish.

Reply:
why ? I am confused. This is just operation name (string) and it does not even have to be function name

Reply:
I think it's due to the following statement:

kill -${SIGKILL} ${PID}

Reply:
still don't get it. My suggestion is to change "vmdk-opsd shutdown"" to "vmdk-opsd kill" (i.e. replace the command name passed to vmdk_opsd. How does it imact anything else ? No need to have a function kill() {} , the function can be named force_kill() :-)

Reply:
I see. Will try as you suggested.